### PR TITLE
Add suppliers section

### PR DESCRIPTION
**Pull request by NadieKir**
- Add product suppliers table
- Implement adding supplier by choosing existing one
- Implement adding supplier by creating new one
- Implement deleting suppliers

### CODE REVIEW

File: webapp/api/countriesService.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+sap.ui.define([], function () {
```
Comment (Nastassia24) [inline webapp/api/countriesService.js:1]: test comment

File: webapp/api/countriesService.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   *
+   * @async
+   *
+   * @param {string} sUrl - Url to fetch.
+   * @param {Object} [oRequestOptions = {}] - Request options.
+   * @param {function} fErrorHandler - Error handler.
+   *
+   * @returns {*} - Fetched data or error handler result.
+   */
+  async function handleFetch(sUrl, oRequestOptions = {}, fErrorHandler) {
```
Comment (Nastassia24) [inline webapp/api/countriesService.js]: why do you define 3 parameters, but when you call that function, you pass always 1 param. it means that oRequestOptions and fErrorHandler are not used

File: webapp/controller/BaseController.js [hunk 3]
```diff
@@ -109,6 +118,87 @@ sap.ui.define(
...
+          case "sap.m.DateRangeSelection":
+            const sDateRangeSelectionStartValue = oField.getDateValue();
+            const sDateRangeSelectionEndValue = oField.getSecondDateValue();
+
+            return sDateRangeSelectionStartValue === null
+              ? null
+              : { startDate: sDateRangeSelectionStartValue, endDate: sDateRangeSelectionEndValue };
+
+          default:
+            return undefined;
```
Comment (Nastassia24) [inline webapp/controller/BaseController.js]: you don't need return undefined implicitly

File: webapp/controller/BaseController.js [hunk 4]
```diff
@@ -109,6 +118,87 @@ sap.ui.define(
...
+
+        return bAreFieldsFilled;
+      },
+
+      /**
+       * Remove dialog from parent's dependents aggregation.
+       *
+       * @param {sap.ui.core.Fragment} oDialog - Dialog to remove from dependents.
+       */
+      removeDependentDialog: function (oDialog) {
```
Comment (Nastassia24) [inline webapp/controller/BaseController.js:236]: this method is only used in ProductDetailsPage

**Review by Nastassia24 — CHANGES_REQUESTED** [review]

**Review by SamusNiko — APPROVED** [review]

**Pull request merged**